webapp/src/containers/SwapPage/index.tsx

```diff
@@ -302,8 +302,8 @@ const SwapPage: React.FunctionComponent<SwapPageProps> = (
       ...formState,
       [field]:
         formState.hash1 === '0'
-          ? Math.max(Number(value) - 1, 0).toFixed(8)
-          : Number(value).toFixed(8),
+          ? Math.max(new BigNumber(value).toNumber() - 1, 0).toFixed(8)
+          : new BigNumber(value).toFixed(8),
```

Comment:
Can you please find a max function that works on BigNumber, otherwise just use `num1.gt(num2) ? num1 : num2` or something similar